pykoi/rlhf/supervised_finetuning.py

```diff
@@ -83,13 +90,23 @@ def __init__(self, rlhf_config: RLHFConfig, enable_telemetry: bool = True) -> No
             load_in_8bit=self._rlhf_config.load_in_8bit,
             device_map=self._rlhf_config.device_map,
         )
+        # resize the token embeddings to include the added special tokens
+        self.model.resize_token_embeddings(len(self.tokenizer))
+
+        # dh: try the customized data collator that only predicts the answer part
+        data_collator = DataCollatorForCompletionOnlyLM(
+        tokenizer=self.tokenizer, mlm=False, return_tensors="pt", pad_to_multiple_of=8
+        )
+
         self.trainer = SFTTrainer(
             model=self.model,
             args=self.training_args,
             train_dataset=self.dataset["train"],
             eval_dataset=self.dataset["eval"],
-            peft_config=self._rlhf_config.lora_config_rl,
+            peft_config=self._rlhf_config.lora_config_rl, ## TODO: DH: LoraConfig MAY BE IGNORED IF USING FROM_PRETRAINED
             packing=True,
+            data_collator=data_collator,
```

Comment:
qq: could you please help explain in the PR description why we added this `data_collator` while we do not need this before.

Reply:
I got this that this is for training the instruction following objective by masking out the query instead of the casual language model objective for only the next token.

Reply:
okay